util/reggen/uvm_reg_base.sv.tpl

```diff
@@ -122,20 +167,12 @@ ${instantiate_register(reg_width, reg_block_path, r, reg_type, r.name.lower())}\
   % if any_regwen:
       // assign locked reg to its regwen reg
     % for r in rb.all_regs:
-      % if isinstance(r, MultiRegister):
 <%
-      r0 = r.reg if isinstance(r, MultiRegister) else r
+    inst_regs = get_inst_to_reg_dict(r)
 %>\
-        % for idx, reg in enumerate(r.regs):
-<%
-        inst_base = r0.name.lower()
-        inst_name = f'{inst_base}[{idx}]' if len(r.regs) > 1 else inst_base
-%>\
-${apply_regwen(rb, reg, inst_name)}\
-        % endfor
-      % else:
-${apply_regwen(rb, r, r.name.lower())}\
-      % endif
+      % for inst in inst_regs:
+${apply_regwen(rb, inst_regs[inst], inst)}\
+      % endfor
```

Comment:
I like this! But it can be collapsed even further now:
```
    % for r in rb.all_regs:
      % for inst, reg in get_inst_to_reg_dict(r).items():
${apply_regwen(rb, reg, inst)}\
      % endfor
    % endfor
```

Reply:
Good idea. Fixed.